lib/merchant.rb

```diff
@@ -107,4 +107,19 @@ def customers_with_pending_invoices
       invoice.customer
     end
   end
+
+  def quantity_successful_items
+    merchant_transactions = invoices.map do |invoice|
+      invoice.transactions
+    end.flatten
+    successful_transactions = merchant_transactions.select do |transaction|
+        transaction.successful?
+    end
+    successful_invoices = successful_transactions.map do |transaction|
+      transaction.invoice
+    end
+    successful_invoice_items = successful_invoices.map do |invoice|
```

Comment:
Useless assignment to variable - `successful_invoice_items`. Did you mean `successful_invoices`?<br>Pass `&:items` as an argument to `map` instead of a block.